Services/TodoService.cs

```diff
@@ -0,0 +1,346 @@
+
+            Assert.Equal(servicedTodo, todo);
+        }
+
+        [Fact]
+        public void GetById_CallRepositoryGetById()
+        {
+            string guid = "00000000-0000-0000-0000-000000000001";
+            MockRepository.Setup(a => a.GetById(Guid.Parse(guid)));
+
+            var service = new Service.TodoService(MockRepository.Object);
+            service.GetByID(Guid.Parse(guid));
+
+            MockRepository.Verify(a => a.GetById(Guid.Parse(guid)), Times.Once);
+        }
+
+        [Theory]
+        [InlineData("0a000300-0600-0000-0100-0000f0700001", "Picnic", "Go to a picnic with friends", TodoStatus.Open, "2020-05-15T14:29:15.1823029Z", "2020-05-19T21:00:00.0000000Z")]
+        public void GivenExistingTodo_Update_UpdateRetursTrue(string id, string title, string description, TodoStatus status, string createdOn, string dueDate)
+        {
+            var todo = new TodoModel
+            {
+                Id = Guid.Parse(id),
+                Title = title,
+                Description = description,
+                Status = status,
+                CreatedOn = DateTime.Parse(createdOn),
+                DueDate = DateTime.Parse(dueDate)
+            };
+
+            MockRepository.Setup(a => a.Update(todo));
+            todo.Title = "Concert";
+            todo.Description = "Go to Metallica concert";
+            var service = new Service.TodoService(MockRepository.Object);
+
+            var isUpdated = service.Update(todo);
+
+            Assert.True(isUpdated);
+        }
+
+        [Fact]
+        public void Update_CallRepositoryUpdate()
+        {
+            var todo = new TodoModel();
+            MockRepository.Setup(a => a.Update(todo));
+
+            var service = new Service.TodoService(MockRepository.Object);
+            service.Update(todo);
+
+            MockRepository.Verify(a => a.Update(todo), Times.Once);
+        }
+
+        [Fact]
+        public void Create_CallRepositoryInsert()
+        {
+            var todo = new TodoModel();
+            MockRepository.Setup(a => a.Insert(todo));
+
+            var service = new Service.TodoService(MockRepository.Object);
+            service.Create(todo);
+
+            MockRepository.Verify(a => a.Insert(todo), Times.Once);
+        }
+
+        [Fact]
+        public void Create_CallRepositorySave()
+        {
+            var todo = new TodoModel();
+            MockRepository.Setup(a => a.Save());
+
+            var service = new Service.TodoService(MockRepository.Object);
+            service.Create(todo);
+
+            MockRepository.Verify(a => a.Save(), Times.Once);
+        }
+
+        [Theory]
+        [InlineData("0a000300-0600-0000-0100-0000f0700001", "Picnic", "Go to a picnic with friends", TodoStatus.Open, "2020-05-15T14:29:15.1823029Z", "2020-05-19T21:00:00.0000000Z")]
+        public void GivenCorrectTodo_Create_ReturnCorrectTodo(string id, string title, string description, TodoStatus status, string createdOn, string dueDate)
```

Comment:
`TodoService.Create()` should return the Todo that is returned from `TodoRepository.Insert()`.
The returned todo might be different from the one provided as argument.

You need to update the test. It will fail, cause the service is returning the correct todo model. You need to fix it.